src/source/azuread.ts

```diff
@@ -58,18 +66,31 @@ class AzureAD {
 
   private async getEnterpriseApplications(
     accessToken: string,
-    url = 'https://graph.microsoft.com/v1.0/applications'
+    serviceNames: string[] = [],
+    baseURL = 'https://graph.microsoft.com/v1.0/applications'
   ) {
+    const filterQueries = serviceNames.map(name => `(displayName eq '${name}')`).join(' or ');
+    const url = filterQueries ? `${baseURL}?$filter=${filterQueries}` : baseURL;
```

Comment:
I'd use the URL class here.

```suggestion
    const url = new URL(baseURL);
    if (filterQueries) {
      url.searchParams.set('$filter', filterQueries);
    }
```

Reply:
I fixed it 👏